src/app/space/add-app-overlay/add-app-overlay.component.spec.ts

```diff
@@ -327,6 +327,35 @@ describe('AddAppOverlayComponent', () => {
       });
     });
 
+    it('continue button should be disabled on navigation in progress', async(() => {
+      const element: HTMLElement = fixture.debugElement.nativeElement;
+      let btnElem = element.querySelector('.code-imports--step_toolbar > button');
+      expect(btnElem.hasAttribute('disabled')).toBeTruthy();
+      component.navigationInProgress = false;
+      component.projectName = 'project-aug-16-2018';
+      component.selectedFlow = 'createapp';
+      component.validateProjectName();
+      component.navigationInProgress = true;
+      fixture.detectChanges();
+      fixture.whenStable().then(() => {
+        expect(btnElem.hasAttribute('disabled')).toBeTruthy();
+      });
+    }));
+
+    it('continue button should be enable on navigation is not in progress', async(() => {
+      const element: HTMLElement = fixture.debugElement.nativeElement;
+      let btnElem = element.querySelector('.code-imports--step_toolbar > button');
+      expect(btnElem.hasAttribute('disabled')).toBeTruthy();
+      component.navigationInProgress = false;
+      component.projectName = 'project-aug-16-2018-1';
+      component.selectedFlow = 'createapp';
+      component.validateProjectName();
+      component.isProjectNameAvailable = true;
+      fixture.detectChanges();
+      fixture.whenStable().then(() => {
+        expect(btnElem.hasAttribute('disabled')).toBeFalsy();
+      });
+    }));
```

Comment:
In general - how button tests are related to `add-app-overlay` fix?

Reply:
I added the disabled property to the button that's why I wrote this test. So, When the user clicks on continue button then button get disabled because it takes some time to navigate.